community/kernel/src/main/java/org/neo4j/kernel/impl/factory/DataSourceModule.java

```diff
@@ -168,6 +170,11 @@ public DataSourceModule( final GraphDatabaseFacadeFactory.Dependencies dependenc
         DbmsOperations dbmsOperations = new NonTransactionalDbmsOperations( procedures );
         deps.satisfyDependency( dbmsOperations );
 
+        for ( SessionManager candidate : Service.load( SessionManager.class ) )
```

Comment:
Also, service loading can in principle return multiple candidates in an unpredictable order, and this loop calls `satisfyDependency` for each of them.